.pre-commit-config.yaml

```diff
@@ -0,0 +1,62 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0  # Use the ref you want to point at
+    hooks:
+    -   id: trailing-whitespace
+    -   id: no-commit-to-branch
+        args: ['--branch', 'master']
+    -   id: check-merge-conflict
+    -   id: double-quote-string-fixer
+    -   id: requirements-txt-fixer
+    -   id: check-yaml
+    -   id: check-docstring-first
+    -   id: end-of-file-fixer
+    -   id: name-tests-test
+    -   id: check-ast
+    -   id: debug-statements
+    # -   id: ...
+
+-   repo: https://github.com/PyCQA/flake8
+    rev: 6.0.0
+    hooks:
+    -   id: flake8
+
+-   repo: https://github.com/asottile/add-trailing-comma
+    rev: v2.4.0
+    hooks:
+    -   id: add-trailing-comma
```

Comment:
It is usually better to keep hooks that mutate files on disk at the beginning, so the following checks would act on the already formatted files.
Let's move it to the beginning of the config.

Reply:
Is it worth moving up the pylint hook too?

Reply:
pylint is not a formatter, plus it tends to be heavy/slow so I often place it towards the end so that the pre-commit run is as responsive as possible.
It should be visible in the example I linked yesterday.

Reply:
Resolved.